pqos/alloc.c

```diff
@@ -1057,13 +1057,18 @@ void alloc_print_config(const struct pqos_capability *cap_mon,
                         const struct pqos_capability *cap_l3ca,
                         const struct pqos_capability *cap_l2ca,
                         const struct pqos_capability *cap_mba,
-                        const unsigned sock_count,
-                        const unsigned *sockets,
                         const struct pqos_cpuinfo *cpu_info,
                         const int verbose)
 {
         int ret;
         unsigned i;
+	unsigned sock_count, *sockets = NULL;
+
+	sockets = pqos_cpu_get_sockets(cpu_info, &sock_count);
```

Comment:
Memory leak - socekts are not always freed

Reply:
Good catch.  Fixed it now. Thanks